src/operations/Completion.cpp

```diff
@@ -234,38 +281,63 @@ void WorkspaceFolder::suggestImports(
         for (auto& service : services)
         {
             // ASSUMPTION: if the service was defined, it was defined with the exact same name
-            bool isAlreadyDefined = false;
+            if (visitor.serviceLineMap.find(service) != visitor.serviceLineMap.end())
+                continue;
+
             size_t lineNumber = minimumLineNumber;
             for (auto& [definedService, stat] : visitor.serviceLineMap)
             {
                 auto location = stat->location.begin.line;
-                if (definedService == service)
-                {
-                    isAlreadyDefined = true;
-                    break;
-                }
-
                 if (definedService < service && location >= lineNumber)
                     lineNumber = location + 1;
             }
 
-            if (isAlreadyDefined)
-                continue;
+            result.emplace_back(createSuggestService(service, lineNumber));
+        }
+    }
 
-            auto importText = "local " + service + " = game:GetService(\"" + service + "\")\n";
+    {
+        FindRequiresVisitor visitor;
+        visitor.visit(sourceModule->root);
 
-            lsp::CompletionItem item;
-            item.label = service;
-            item.kind = lsp::CompletionItemKind::Class;
-            item.detail = "Auto-import";
-            item.documentation = {lsp::MarkupKind::Markdown, codeBlock("lua", importText)};
-            item.insertText = service;
-            item.sortText = SortText::AutoImports;
+        for (auto& [path, node] : fileResolver.virtualPathsToSourceNodes)
+        {
+            if (path == moduleName || node->className != "ModuleScript" || visitor.contains(node->name))
+                continue;
+            auto scriptFilePath = node->getScriptFilePath();
+            if (scriptFilePath.has_value() && isIgnoredFile(scriptFilePath.value(), config))
```

Comment:
Need to use `WorkspaceFileResolver::getRealPathFromSourceNode()` here, since it handles correct relative paths
```suggestion
            if (auto scriptFilePath = fileResolver.getRealPathFromSourceNode(node); scriptFilePath && isIgnoredFile(scriptFilePath.value(), config))
```